DataStructures/Lists/CircularBuffer.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+
+namespace DataStructures.Lists {
+    public class CircularBuffer<T> {
+        /// <summary>
+        /// Initializes a circular buffer with initial length of 10
+        /// </summary>
+        public CircularBuffer(bool canOverride=true) : this(10,canOverride) {
```

Comment:
Can we add property and use it instead that hardcoded 10
        private static readonly int defaultBufferLength = 10;
This allow us avoid magic numbers in code.

Reply:
Yes we can. I added a private static readonly field to hold the default buffer length. Since properties are meant to be public.